File: typescript/interpreter/errorReporter.ts
export let hadError = false;

export function resetError() {
  hadError = false;
}

export function reportError(line: number, message: string) {
  report(line, "", message);
}

function report(line: number, where: string, message: string) {
  console.error(`[line ${line}] Error ${where}: ${message}`);
  hadError = true;
}

File: python/interpreter/error_reporter.py
class ErrorReporter:
    def __init__(self):
        self.had_error = False

    def error(self, line: int, message: str):
        self.report(line, "", message)

    def report(self, line: int, where: str, message: str):
        print("[line " + str(line) + "] Error " + where + ": " + message)
        self.had_error = True
